src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -106,7 +106,7 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                             // so we read the secrets running them through the appropriate readers
                             hostSecrets = ReadHostSecrets(hostSecrets);
                         }
-                        catch (CryptographicException ex)
+                        catch (CryptographicException ex) when (!ex.InnerException.IsFatal())
```

Comment:
Now, if we have a failure here, hostSecrets will be null, which will lead to a null ref below on `hostSecrets.HasStaleKeys`. It seems we'll need special handling in these methods for the case where we can't read keys at all, and tests to ensure we handle the lack of keys (nulls) properly.